lib/Core/Executor.cpp

```diff
@@ -436,16 +436,27 @@ const char *Executor::TerminateReasonNames[] = {
   [ Unhandled ] = "xxx",
 };
 
-
-Executor::Executor(LLVMContext &ctx, const InterpreterOptions &opts,
-                   InterpreterHandler *ih)
+static int tries = 0;
+bool enteredDevMain = false;
+Executor::Executor(LLVMContext &ctx,
+	           const InterpreterOptions &opts,
+                   InterpreterHandler *ih,
+		   						 TimingSolver* s,
+									 KModule* _kmodule,
+									 SpecialFunctionHandler *_specialFunctionHandler,
+									 StatsTracker *_statsTracker)
     : Interpreter(opts), interpreterHandler(ih), searcher(0),
-      externalDispatcher(new ExternalDispatcher(ctx)), statsTracker(0),
-      pathWriter(0), symPathWriter(0), specialFunctionHandler(0), timers{time::Span(TimerInterval)},
+      externalDispatcher(new ExternalDispatcher(ctx)), /*statsTracker(0),*/
+      pathWriter(0), symPathWriter(0), /*specialFunctionHandler(0),*/ timers{time::Span(TimerInterval)},
       replayKTest(0), replayPath(0), usingSeeds(0),
       atMemoryLimit(false), inhibitForking(false), haltExecution(false),
       ivcEnabled(false), debugLogBuffer(debugBufferString) {
 
+	enteredDevMain = false; // Must be reset; previous Executor instance will have set it to true, leading to problems with undefined memory in uclibc before user main is reached.
```

Comment:
what's the error message you get?

Reply:
If I comment this line, it will print
"addConstraint1: offset=2undefined_offset=2
KLEE: ERROR: libc/string/memcpy.c:29: read non-initialized memory"
 for the second input.
I think the reason is in this line: https://github.com/leonbett/klee/blob/debug2/lib/Core/Memory.cpp#L378

Reply:
yes. you need to install my patched [uclibc](https://github.com/lmrs2/klee-uclibc). Once I added support of non-initialized read in KLEE, it started finding some bugs in uclibc itself :/

Reply:
Interesting :)
I've reverted my changes and used your uclibc, but then it's detecting another undefined read:
KLEE: ERROR: /root/git/klee/runtime/POSIX/fd_init.c:78: read non-initialized memory, which is this line klee_prefer_cex(s, s->st_dev == defaults->st_dev);

Reply:
yeah, that's because I have not tested all possible options... :/ If you check https://github.com/klee/klee/blob/master/runtime/POSIX/fd_init.c#L117, you'll see that `defaults->st_dev` is never set, so KLEE thinks it's undefined. The call to `stat64()` calls into kernel and initializes the `s` variable, but KLEE is not aware of this.
So we need to tell KLEE to ignore the undefinedness and use the values as concrete values. This can be done with `klee_ignore_undefined()`, see how it's used in https://github.com/lmrs2/klee-uclibc/blob/619421b12979bcea0b838338df37f23b3af46e8e/libc/termios/tcgetattr.c
This should work. If it does, I'll patch my KLEE fork...

Reply:
Thanks, [this](https://github.com/leonbett/klee/pull/1/commits/ae73e8ea54d9225b7600a1e829200cd5e57ad91f) did the trick.